assets/chef-recipes/cookbooks/mariadb/recipes/uninstall.rb

```diff
@@ -1,19 +1,49 @@
+service 'mariadb' do
+  action :stop
+end
+
+execute 'Reset iptables settings' do
+  command 'iptables -D INPUT -p tcp --dport 3306 -j ACCEPT -m state --state NEW'
+end
+
 case node[:platform_family]
-when "debian"
-  package "mariadb-common" do
+when 'debian'
+  package 'mariadb-common' do
+    action :purge
+  end
+  file '/etc/apt/sources.list.d/mariadb.list' do
+    action :delete
+  end
+  apt_update 'update apt cache' do
+    action :update
+  end
+  execute 'Save iptables settings' do
+    command 'iptables-save > /etc/iptables/rules.v4'
+  end
+
+when 'rhel', 'centos'
+  package 'MariaDB-common' do
     action :remove
   end
-  execute "Remove mariadb repository" do
-    command "rm -fr /etc/apt/sources.list.d/mariadb.list"
+  file '/etc/yum.repos.d/mariadb.repo' do
+    action :delete
   end
-  execute "update" do
-    command "apt-get update"
+  execute 'Save iptables settings' do
+    if node[:platform_version].to_i == 6
+      command '/sbin/service iptables save'
+    else
+      command 'iptables-save > /etc/sysconfig/iptables'
+    end
   end
-when "rhel", "fedora", "suse"
-  package "MariaDB-common" do
+
+when 'suse'
+  package 'MariaDB-common' do
     action :remove
   end
-  execute "Remove repo" do
-    command "rm -fr /etc/yum.repos.d/mariadb.repo /etc/zypp/repos.d/mariadb.repo*"
+  file '/etc/zypp/repos.d/mariadb.repo*' do
```

Comment:
Zypper repository has the remove action, https://docs.chef.io/resources/zypper_repository/